### PR TITLE
Ethereum transactions identifier + optimizations on the combined trades view

### PR DESCRIPTION
**Pull request by LefterisJP**
- Adds identifier to the ethereum transactions response
- Various optimizations for the combined trades view

### CODE REVIEW

File: rotkehlchen/db/schema.py [hunk 1]
```diff
@@ -724,19 +724,19 @@
         WHERE first1in > 0 AND first0in > 0 AND last1out > 0 AND last0out > 0
     ), SWAPS AS (
     SELECT base1 AS base_asset, quote1 AS quote_asset, amount1 AS amount, rate1 AS rate, txhash, logindex, time, location FROM C1
-    UNION
+    UNION ALL /* using union all as there can be no duplicates so no need to handle them */
     SELECT base1 AS base_asset, quote1 AS quote_asset, amount1 AS amount, rate1 AS rate, txhash, logindex, time, location FROM C2
-    UNION
+    UNION ALL /* using union all as there can be no duplicates so no need to handle them */
```
Comment (yabirgb) [inline rotkehlchen/db/schema.py:729]: Does `UNION ALL` perform better than only `UNION` in the case of no duplicates? Couldn't find anything in the docs here https://www.sqlite.org/optoverview.html

Comment (LefterisJP) [reply rotkehlchen/db/schema.py:729]: UNION is a lot more expensive an operation than UNION ALL as it also sorts unioned table and removes duplicates.

Comment (LefterisJP) [reply rotkehlchen/db/schema.py:729]: https://stackoverflow.com/a/3627970/110395

**Comment by LefterisJP** [comment]
The only test failure was a timeout during a contract call.

```
2021-12-23T23:18:59.4930020Z WARNING  rotkehlchen.chain.ethereum.manager:manager.py:418 Greenlet-7: Failed to query etherscan for <bound method EthereumManager._call_contract of <rotkehlchen.chain.ethereum.manager.EthereumManager object at 0x7f47c9492a50>> due to Etherscan API request failed due to HTTPSConnectionPool(host='api.etherscan.io', port=443): Read timed out. (read timeout=30)
2021-12-23T23:18:59.4935494Z DEBUG    rotkehlchen.api.rest:rest.py:178 Greenlet-7: Request successful response={"result": null, "message": "Failed to query <bound method EthereumManager._call_contract of <rotkehlchen.chain.ethereum.manager.EthereumManager object at 0x7f47c9492a50>> after trying the following nodes: ['blockscout', '1inch', 'mycrypto', 'etherscan', 'cloudflare-eth', 'myetherwallet', 'avado pool']. Check logs for details."}, 
```

Unrelated to the PR, so merging.

**Pull request merged**